packages/file_selector/file_selector_windows/windows/file_dialog_controller.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "file_dialog_controller.h"
+
+#include <windows.h>
+
+namespace file_selector_windows {
+
+FileDialogController::FileDialogController(IFileDialog* dialog)
+    : dialog_(dialog) {
+  dialog_->AddRef();
```

Comment:
Nothing to do here, but I wonder if long-term it might be useful to use [ComPtr](https://docs.microsoft.com/en-us/cpp/cppcx/wrl/comptr-class?view=msvc-170) for these sorts of cases -- this one and the local below.